util/topgen/merge.py

```diff
@@ -852,127 +851,187 @@ def amend_reset_request(topcfg: OrderedDict,
-            # only in dio_modules
-            raise SystemExit(
-                "Curren version doesn't support signal in mio_modules {}".
-                format(e))
+            sig_list = deepcopy(block.get_signals_as_list_of_dicts())
+
+            for sig_inst in sig_list:
+                if sig_inst['width'] > 1:
+                    for idx in range(sig_inst['width']):
+                        sig_inst_copy = deepcopy(sig_inst)
+                        sig_inst_copy.update({'idx': idx,
+                                              'pad': sig['pad']})
+                        sig_inst_copy['name'] = sig['name'] + '_' + sig_inst_copy['name']
+                        append_io_signal(temp, sig_inst_copy)
+                else:
+                    sig_inst.update({'idx': -1,
+                                     'pad': sig['pad']})
+                    sig_inst['name'] = sig['name'] + '_' + sig_inst['name']
+                    append_io_signal(temp, sig_inst)
+
+    # Now that we've collected all input and output signals,
+    # we can go through once again and stack them into one unified
+    # list, and calculate MIO/DIO global indices.
+    pinmux['ios'] = (temp['inouts'] +
+                     temp['inputs'] +
+                     temp['outputs'])
+
+    # Remember these counts to facilitate the RTL generation
+    pinmux['io_counts'] = {'dedicated': {'inouts': 0, 'inputs': 0, 'outputs': 0, 'pads': 0},
+                           'muxed': {'inouts': 0, 'inputs': 0, 'outputs': 0, 'pads': 0}}
+
+    for sig in pinmux['ios']:
+        glob_idx = get_index_and_incr(pinmux['io_counts'], sig['pad'], sig['type'])
+        sig.update({'glob_idx': glob_idx})
+
+    # Calculate global indices for pads.
+    j = k = 0
+    for pad in pinout['pads']:
+        if pad['muxed']:
+            pad.update({'idx': j})
+            j += 1
+        else:
+            pad.update({'idx': k})
+            k += 1
+    pinmux['io_counts']['muxed']['pads'] = j
+    pinmux['io_counts']['dedicated']['pads'] = k
+
+    # For each target configuration, calculate the special signal indices.
+    known_muxed_pads = {}
+    for pad in pinout['pads']:
+        if pad['muxed']:
+            known_muxed_pads.update({pad['name']: pad})
+
+    known_mapped_dio_pads = {}
+    for sig in pinmux['ios']:
+        if sig['pad'] in ['muxed', 'custom']:
+            continue
+        if sig['pad'] in known_mapped_dio_pads:
+            raise SystemExit('Cannot have multiple IOs mapped to the same DIO pad {}'
+                             .format(sig['pad']))
+        known_mapped_dio_pads.update({sig['pad']: sig})
+
+    for target in targets:
+        for entry in target['special_pads']:
+            # If this is a muxed pad, the resolution is
+            # straightforward. I.e., we just assign the MIO index.
+            if entry['pad'] in known_muxed_pads:
+                entry.update({'idx': known_muxed_pads[entry['pad']]['idx']})
```

Comment:
`entry['idx'] = known_muxed_pads[entry['pad']]['idx']` should be equivalent?

Reply:
yes - changed